### PR TITLE
Popup login and register UX design and feature changes

### PR DESCRIPTION
**Pull request by aleixac**
Added interfaces and code for the new login and register related endpoints:
Password reset, password change, and logout through endpoint.
Adapted also the texts, html and styles, to the UX login and register design inputs.
Issues linked:
- Change Password: https://github.com/co-inform/browser-plugin/issues/15
- Forgot Password: https://github.com/co-inform/browser-plugin/issues/14
- Register procedure: https://github.com/co-inform/browser-plugin/issues/13

### CODE REVIEW

File: src/plugin/_locales/en/messages.json [hunk 1]
```diff
@@ -189,7 +192,7 @@
     "message": "Something went wrong during the registration. Try again"
   },
   "register_ok": {
-    "message": "User succesfully registered. Now you can log in"
+    "message": "User succesfully registered. You will receive an email with instruccions to validate your account. You need to validate the account prior to log in."
```
Comment (daffyDukk) [inline src/plugin/_locales/en/messages.json:195]: instructions not instruccions

Comment (aleixac) [reply src/plugin/_locales/en/messages.json:195]: thx! it's a catalan typo.. ("instruccions" is catalan word for "instructions") :P

Comment (aleixac) [reply src/plugin/_locales/en/messages.json]: fixed! now I don't know why it says that I dismissed your review.. sorry, was not my intention when I did the typo fix commit..

Comment (daffyDukk) [reply src/plugin/_locales/en/messages.json]: no worries.!

**Pull request merged**